netbox/dcim/forms/bulk_edit.py

```diff
@@ -285,17 +285,39 @@ class RackBulkEditForm(NetBoxModelBulkEditForm):
         widget=SmallTextarea,
         label='Comments'
     )
+    weight = forms.DecimalField(
+        min_value=0,
+        required=False
+    )
+    weight_unit = forms.ChoiceField(
+        choices=add_blank_choice(DeviceWeightUnitChoices),
+        required=False,
+        initial='',
+        widget=StaticSelect()
+    )
 
     model = Rack
     fieldsets = (
         ('Rack', ('status', 'role', 'tenant', 'serial', 'asset_tag')),
         ('Location', ('region', 'site_group', 'site', 'location')),
         ('Hardware', ('type', 'width', 'u_height', 'desc_units', 'outer_width', 'outer_depth', 'outer_unit')),
+        ('Attributes', ('weight', 'weight_unit')),
     )
     nullable_fields = (
-        'location', 'tenant', 'role', 'serial', 'asset_tag', 'outer_width', 'outer_depth', 'outer_unit', 'comments',
+        'location', 'tenant', 'role', 'serial', 'asset_tag', 'outer_width', 'outer_depth', 'outer_unit', 'comments', 'weight'
     )
 
+    def clean(self):
+        super().clean()
+
+        # Validate weight/unit
+        weight = self.cleaned_data.get('weight')
+        weight_unit = self.cleaned_data.get('weight_unit')
+        if weight and not weight_unit:
+            raise forms.ValidationError({
+                'weight_unit': "Must specify a unit when setting weight"
+            })
+
```

Comment:
I don't believe this is needed; validation should be handled by the model's `clean()` method.
```suggestion
```